src/RpcClient/TransactionManager.cs

```diff
@@ -197,7 +183,8 @@ public TransactionManager AddWitness(UInt160 scriptHash, params object[] paramet
         public TransactionManager Sign()
         {
             // Calculate NetworkFee
-            Tx.NetworkFee = CalculateNetworkFee();
+            Tx.NetworkFee = new DummyWallet().CalculateNetworkFee(Blockchain.Singleton.GetSnapshot(), Tx);
```

Comment:
it will cause timeout because Blockchain can not be initialized in rpcclient if without full-ledger data.

Reply:
Without the ledger data we can't compute the fee....

Reply:
We need snapshot to run the contracts. Otherwise it is impossible to compute the fees.

Reply:
> it will cause timeout because Blockchain can not be initialized in rpcclient if without full-ledger data.

@shargon  for rpcclient, the Blockchain.Singleton is null. 

```cs
        public static LocalNode Singleton
        {
            get
            {
                while (singleton == null) Thread.Sleep(10);
                return singleton;
            }
        }
```

Reply:
Can we calculate the network fee in RpcServer?

Reply:
We can send the tx in raw and compute the fee

Reply:
@cloud8little please check it again